eng/tools/smoketests/main.go

```diff
@@ -352,6 +350,76 @@ func BuildMainFile(root string) error {
 	return err
 }
 
+func FindEnvVars(root string) error {
+	fmt.Println("Find all environment variables using `os.Getenv` or `os.LookupEnv`")
+
+	err := filepath.Walk(root, func(path string, info fs.FileInfo, err error) error {
+		if strings.HasSuffix(path, ".go") {
+			// Find Env Vars
+			searchFile(path)
+		}
+		return nil
+	})
+	return err
+}
+
+func searchFile(path string) error {
+	var envVarRegex *regexp.Regexp
+	if runtime.GOOS == "windows" {
+		envVarRegex = regexp.MustCompile(`(?m)os.LookupEnv(.*)\r\n`)
```

Comment:
Is the newline match here necessary? I could see it excluding potential future scenarios like:

```
if val, ok = os.LookupEnv("foo"); !ok {
...
```

Why not match inside the parentheses+quotes instead?

```suggestion
		envVarRegex = regexp.MustCompile(`(?m)os.LookupEnv\("(.*)"\)`)
```